src/main/java/seedu/address/logic/parser/SearchCommandParser.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.Arrays;
+import java.util.function.Predicate;
+
+import seedu.address.logic.commands.SearchCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.NameContainsKeywordsPredicate;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.TagContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindCommand object
+ */
+public class SearchCommandParser implements Parser<SearchCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a SearchCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SearchCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, CliSyntax.PREFIX_NAME, CliSyntax.PREFIX_TAG);
+
+        // Checks for multiple instances of the same prefix
+        checkForMultipleInstances(argMultimap, CliSyntax.PREFIX_NAME, "name");
+        checkForMultipleInstances(argMultimap, CliSyntax.PREFIX_TAG, "tag");
+
+        String nameArgs = argMultimap.getValue(CliSyntax.PREFIX_NAME).orElse("");
+        String tagArgs = argMultimap.getValue(CliSyntax.PREFIX_TAG).orElse("");
+
+        // Check for empty inputs after the prefixes
+        checkForEmptyInput(argMultimap, CliSyntax.PREFIX_NAME, nameArgs, "name");
+        checkForEmptyInput(argMultimap, CliSyntax.PREFIX_TAG, tagArgs, "tag");
```

Comment:
Is it possible to make use of the classes: `Name` and `Tag`  here? Then, would it be possible to utilise methods from `ParserUtil` such as `parseName()` and `parseTags()`.

Do let me know what you think 😄

Reply:
I don't think it is needed to utilise parseName() and parseTags() as we are working with String values provided as a keyword here for matching. Or did I not catch how you would think parseName() and parseTags() could be used. Do let me know! 👍

Reply:
My intention was for there to not be a duplication of code as aside from parsing the input, `parseName()` and `parseTags()` functions validates if the input `Name` or `Tag` are valid.  

However, now that I think about it, I think that doing so would lead to the `ParseException` error being thrown when the parameters are incorrect. This may not be the intended behaviour if we expect the application to just return that no `Person` was found with such parameters, even with "incorrect" parameters.

As such, I think its fine for now. But side not, I think doing `PREFIX_NAME` instead of `CliSyntax.PREFIX_NAME` would be better?

Reply:
Oh shucks, thank you for pointing the last point out! Will make the necessary changes. Thank you Minlong 😊